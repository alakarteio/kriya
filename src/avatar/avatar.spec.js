/* eslint-disable import/no-extraneous-dependencies, react/jsx-filename-extension */
/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import ReactTestUtils from 'react-addons-test-utils'
import mockComponent from 'misc/__mocks__/component'
import Avatar from './avatar'

jest.mock('../icon', () => mockComponent('icon'))

const component = (props) => {
  return renderer.create(
    <Avatar {...props} />,
  )
}

const snap = (props, Component) => {
  const c = Component || component(props)
  const tree = c.toJSON()
  expect(tree).toMatchSnapshot()
}

describe('common/Icon', () => {
  describe('graphical (JSX)', () => {
    it('should add custom className', () => snap({ className: 'custom' }))
    it('should add custom style', () => snap({ style: { backgroundColor: 'red' } }))
    it('should create a picture avatar', () => snap({ picture: 'picture' }))
    it('should create male avatar', () => snap({ isMale: true }))
    it('should create female avatar', () => snap({ isMale: false }))
    it('should have a default behaviour', () => snap({}))

    it('should print avatar when image is on error', () => {
      const rendered = ReactTestUtils.renderIntoDocument(<Avatar picture="url://url.jpeg" />)
      let img = ReactTestUtils.findRenderedDOMComponentWithTag(rendered, 'img')
      let icon = ReactTestUtils.scryRenderedDOMComponentsWithClass(rendered, 'icon')
      expect(img).toBeDefined()
      expect(icon).toHaveLength(0)

      // Fake error
      ReactTestUtils.Simulate.error(img)

      // Image is not printed anymore, icon is
      img = ReactTestUtils.scryRenderedDOMComponentsWithTag(rendered, 'img')
      icon = ReactTestUtils.findRenderedDOMComponentWithClass(rendered, 'icon')
      expect(img).toHaveLength(0)
      expect(icon).toBeDefined()
    })
  })
})

/* eslint-enable import/no-extraneous-dependencies, react/jsx-filename-extension */
