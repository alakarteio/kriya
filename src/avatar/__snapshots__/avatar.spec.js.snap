// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/Icon graphical (JSX) should add custom className 1`] = `
<div
  className="avatar custom"
  style={Object {}}
>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"icon","primary":true,"children":"avatar_woman","style":{},"alt":"avatar"}}
  </div>
</div>
`;

exports[`common/Icon graphical (JSX) should add custom style 1`] = `
<div
  className="avatar"
  style={
    Object {
      "backgroundColor": "red",
    }
  }
>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"icon","primary":true,"children":"avatar_woman","style":{},"alt":"avatar"}}
  </div>
</div>
`;

exports[`common/Icon graphical (JSX) should create a picture avatar 1`] = `
<div
  className="avatar"
  style={Object {}}
>
  <img
    alt="avatar"
    className="img"
    onError={[Function]}
    src="picture"
    style={Object {}}
  />
</div>
`;

exports[`common/Icon graphical (JSX) should create female avatar 1`] = `
<div
  className="avatar"
  style={Object {}}
>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"icon","primary":true,"children":"avatar_woman","style":{},"alt":"avatar"}}
  </div>
</div>
`;

exports[`common/Icon graphical (JSX) should create male avatar 1`] = `
<div
  className="avatar"
  style={Object {}}
>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"icon","primary":true,"children":"avatar_man","style":{},"alt":"avatar"}}
  </div>
</div>
`;

exports[`common/Icon graphical (JSX) should have a default behaviour 1`] = `
<div
  className="avatar"
  style={Object {}}
>
  <div
    className="icon"
  >
    {"name":"icon","props":{"className":"icon","primary":true,"children":"avatar_woman","style":{},"alt":"avatar"}}
  </div>
</div>
`;
